alica_engine/include/engine/RunningPlan.h

```diff
@@ -238,6 +243,10 @@ class RunningPlan : public std::enable_shared_from_this<RunningPlan>
     std::map<const AbstractPlan*, int> _failedSubPlans;
 
     mutable std::mutex _accessMutex;
+
+    std::shared_ptr<scheduler::Job> _job;
+    std::shared_ptr<scheduler::Job> _terminateJob;
+    std::shared_ptr<scheduler::Job> _initJob;
```

Comment:
job shared pointer should only be in scheduler and it will be destructed once it's executed. Running plan should  only have a weak_ptr